src/Detector/Framework.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Utopia\Detector\Detector;
+
+use Utopia\Detector\Detection\Framework as FrameworkDetection;
+use Utopia\Detector\Detector;
+
+class Framework extends Detector
+{
+    public function __construct(
+        protected array $inputs,
+        protected string $packager = 'npm'
+    ) {}
+
+    public function detect(): ?FrameworkDetection
+    {
+        foreach ($this->options as $detector) {
+            $detectorFiles = $detector->getFiles();
+
+            foreach ($this->inputs as $input) {
+                foreach ($detectorFiles as $pattern) {
+                    if ($input === $pattern) {
```

Comment:
I think there is some inconsistency between detect method of different detections.
Here we do nested loops, packager does array_intersect, and rendering does nested foreach too.

Let's find out what is the best strategy and stick with that everywhere for consistency. Pinging engineers on Discord can help us find out if array_intersect is the fastest method, or not.